### PR TITLE
OpenAI and Bedrock Preview

### PR DESCRIPTION
**Pull request by lrafeei**
# Overview

- Merges Bedrock and OpenAI branches
- Adds ML metrics for OpenAI and Bedrock

### CODE REVIEW

**Comment by umaannamalai** [comment]
We will hold off on merging this PR to main until the features are GAed.

File: tests/external_botocore/test_bedrock_chat_completion.py [hunk 1]
```diff
@@ -123,18 +125,19 @@ def _test():
...
-    #     name="test_bedrock_chat_completion_in_txn_no_convo_id",
-    #     custom_metrics=[
-    #         ("Python/ML/OpenAI/%s" % openai.__version__, 1),
-    #     ],
-    #     background_task=True,
-    # )
+    @validate_transaction_metrics(
+        name="test_bedrock_chat_completion_in_txn_no_convo_id",
+        custom_metrics=[
+            ("Python/ML/Botocore/%s" % BOTOCORE_VERSION, 1),
```
Comment (hmstepanek) [inline tests/external_botocore/test_bedrock_chat_completion.py:137]: This should actually be `Python/ML/Bedrock/%s`

**Review by hmstepanek — CHANGES_REQUESTED** [review]

**Review by hmstepanek — APPROVED** [review]
Looks good!

**Review by umaannamalai — APPROVED** [review]
🚀

**Pull request merged**